### PR TITLE
Docs: Adding docs changes related to the Moment.js dependency update

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Updates Moment.js version numbers mentioned in the docs
- Edits the changelog entry

### CODE REVIEW

File: .changelogs/9381.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "title": "Update MomentJS dependency to 2.29.2.",
+  "title": "Updated the Moment.js dependency from `2.24.0` to `2.29.3`, improving Handsontable's date format matching.",
```
Comment (jansiegel) [inline .changelogs/9381.json:2]: I think I'd rather concentrate on the security aspect of the change, as it's not exactly clear if the change on Moment's side is something we want or not.

File: docs/next/guides/building-and-testing/packages.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ If you are a "Bob the Builder" kind of hacker, you will need to load Handsontabl
 ```html
 <!-- Required dependencies (as external scripts) -->
 <link href="https://cdn.jsdelivr.net/npm/pikaday@1.5.1/css/pikaday.min.css" rel="stylesheet">
-<script src="https://cdn.jsdelivr.net/npm/moment@2.20.1/moment.min.js"></script>
+<script src="https://cdn.jsdelivr.net/npm/moment@2.29.3/moment.min.js"></script>
```
Comment (jansiegel) [inline docs/next/guides/building-and-testing/packages.md:33]: Good catch, I totally forgot it's linked from the cdn in the docs 👍

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: .changelogs/9381.json (hunk 1 above)
Comment (kirszenbaum) [reply .changelogs/9381.json]: @jansiegel - sure (I followed [this ](https://handsoncode.slack.com/archives/C4PV4S33Q/p1650384660407479?thread_ts=1649328227.856499&cid=C4PV4S33Q) but took it out of context)

How about this: [a430fda](https://github.com/handsontable/handsontable/pull/9409/commits/a430fda7605f72ad4b29e85361694d52cbb1d27f)

Comment (jansiegel) [reply .changelogs/9381.json]: > @jansiegel - sure (I followed [this ](https://handsoncode.slack.com/archives/C4PV4S33Q/p1650384660407479?thread_ts=1649328227.856499&cid=C4PV4S33Q) but took it out of context)
> 
> How about this: [a430fda](https://github.com/handsontable/handsontable/pull/9409/commits/a430fda7605f72ad4b29e85361694d52cbb1d27f) (not to sound too alarmist)

Oh, I missed [this](https://handsoncode.slack.com/archives/C4PV4S33Q/p1650384660407479?thread_ts=1649328227.856499&cid=C4PV4S33Q). Still, I'd rather keep the security information in the changelog, and maybe elaborate about the format matching in the release notes/blog post. WDYT?

https://github.com/handsontable/handsontable/pull/9409/commits/a430fda7605f72ad4b29e85361694d52cbb1d27f looks great to me 👍

**Review by jansiegel — APPROVED** [review]

**Pull request merged**